### PR TITLE
CORS

### PR DESCRIPTION
**Pull request by Destructor1702**
_I was having problems with and had to rebase_

---
Added this to default request options, this might enable CORS in most enviroments
```js
const reqOptions = { 'mode': 'cors', headers: { 'Access-Control-Allow-Origin': '*' } };
``` 

Also you can now set a CORS proxy url in an options parameter if above option is not working

Example:
```js
const NewsAPI = require('newsapi');
const newsapi = new NewsAPI('YOUR_API_KEY', { corsProxyUrl: 'https://cors-anywhere.herokuapp.com/' });
```Added this to default request options, this might enable CORS in most enviroments
```js
const reqOptions = { 'mode': 'cors', headers: { 'Access-Control-Allow-Origin': '*' } };
``` 

Also you can now set a CORS proxy url in an options parameter if above option is not working

Example:
```js
const NewsAPI = require('newsapi');
const newsapi = new NewsAPI('YOUR_API_KEY', { corsProxyUrl: 'https://cors-anywhere.herokuapp.com/' });
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 node_modules
 .env
+yarn.lock
+package-lock.json
```
Comment (bzarras) [inline .gitignore:4]: package-lock.json should remain in source control

**Comment by Destructor1702** [comment]
CI automated tests fail because there's no API key in this repo, you can add a secret key on the reposettings

File: .gitignore (hunk 1 above)
Comment (Destructor1702) [reply .gitignore:4]: ok, i did it because it gives a warning when using yarn

**Comment by bzarras** [comment]
Hmm, I already added the secret key, it seems to have worked for other changes that I recently pushed up

**Comment by bzarras** [comment]
Ah, this is why it's failing:
image

**Pull request merged**